File: 6-kyu/Build a pile of Cubes.js
function findNb(m) {
  let sum = 0, n;
  for (n = 0; sum < m; n += 1) {
    sum = Math.pow(n*(n+1), 2) / 4;
  } if (sum === m) return n-1;
  else return (-1);
}

File: 6-kyu/Build a pile of Cubes.java
public class ASum {
	public static long findNb(long m) {
		int n = 0;
        do {
            m = m - (long) Math.pow(n++ + 1d, 3d);
        } while (m > 0);
        return (m == 0) ? n : -1;
	}	
}

